pylele_solid.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+"""
+    Abstract Base and Concrete Classes for all Gugulele Solid objects
+"""
+
+from __future__ import annotations
+
+import os
+import sys
+import argparse
+import datetime
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+from pylele_api import ShapeAPI, Shape
+from pylele_config import Fidelity, Implementation, CARBON, GRAY, LITE_GRAY,DARK_GRAY, ORANGE, WHITE
+
+COLORS = {
+    'carbon': CARBON, 
+    'gray': GRAY, 
+    'lite_gray' : LITE_GRAY,
+    'dark_gray' : DARK_GRAY, 
+    'orange' : ORANGE, 
+    'white': WHITE
+}
+
+def make_or_exist_path(out_path):
+    """ Check a directory exist, and generate if not """
+
+    if not os.path.isdir(out_path):
+        # Path.mkdir(out_path)
+        os.makedirs(out_path)
+
+    assert os.path.isdir(out_path), f"Cannot export to non directory: {out_path}"
+
+def lele_solid_parser(parser=None):
```

Comment:
What is the use case of adding command line parser in this class? Is it to allow it to run independently so to generate a piece of solid?

Reply:
The idea is that the parser is hierarchical: see how the Tube class parser inherits from LeleSolid parameters too: this allows to scale up complex models, only adding the new parameters of interest at a particular abstraction level. Think of Russian Dolls each one adding details to one another.
For instance I would generate a LeleFretboard, a LeleNeck, a LeleBody, etc, ...
Any layer could be called independentely, and finally pylele_main can call all of them as it is the case today.